### ISSUE TITLE
`no-multi-spaces` conflicts with `smart-tabs`

### ISSUE THREAD

**Issue by bluesmoon** [eslint/eslint#2077]
If I have `"no-mixed-spaces-and-tabs":[2, "smart-tabs"]`, then `no-multi-spaces` should not throw an error/warning for lines that use smart-tabs.

test case, try with the following code:
```
/*eslint no-mixed-spaces-and-tabs:[2, "smart-tabs"], no-multi-spaces:2 */
var i=10,
    j=20;
```

**Comment by nzakas** [comment]
What version of ESLint are you using? I don't see any errors with this code.

In the future, please also include the errors you're seeing on the console when reporting an issue. It really helps us triage.

**Comment by bluesmoon** [comment]
0.17.0

```
   441:7   warning  Multiple spaces found before 'i'                  no-multi-spaces
   442:7   warning  Multiple spaces found before 'gotcookies'         no-multi-spaces
   443:7   warning  Multiple spaces found before 'cookies'            no-multi-spaces
```

code:
```
			var cookies_a,
			    i, l, kv,
			    gotcookies=false,
			    cookies={};
```
(3 tabs, and then 4 spaces to align variable names)

**Comment by nzakas** [comment]
Ah okay, GitHub was converting those to spaces.

Yeah, we can probably just ignore spaces that are in indent position.

**Comment by mourner** [comment]
Came here to open the same issue. :+1:

**Comment by gyandeeps** [comment]
@nzakas @ilyavolodin Based on the discussion, a decision has been made but the labels are not yet updated.

**Comment by nzakas** [comment]
Yup, thanks!

**Comment by message** [comment]
+1

**Comment by tape88** [comment]
+1

**Comment by nzakas** [comment]
@message @tape88 thanks for commenting. This issue is marked as "accepted", so it's just waiting for someone to volunteer to work on it.

**Comment by satya164** [comment]
Have the same issue. Anyone working on it yet?

**Comment by IanVS** [comment]
@satya164 It's all yours if you want it. Looks like some tweaking is needed in: https://github.com/eslint/eslint/blob/master/lib/rules/no-multi-spaces.js#L73

**Comment by satya164** [comment]
@IanVS I can work on it, can you point me to documentation for writing rules?

**Comment by IanVS** [comment]
Yep, rule development docs are at: http://eslint.org/docs/developer-guide/working-with-rules

I would suggest to create a test case first which fails, and then update the regex I linked to so that the test passes.  Feel free to hop in the [Gitter room](https://gitter.im/eslint/eslint) if you need more guidance.

**Comment by IanVS** [comment]
@satya164, do you still plan to work on this?

**Comment by oh-ren** [comment]
+1 :) (Still an issue in 1.9.0)

Btw, although somewhat a different discussion, when using smart-tabs, I also have to set `VariableDeclarator` to `0` in my `indent` options, otherwise I get complaints from that :)
It's my feeling when using smart-tabs, this should be handled in a smart way too. Setting to `0` is of course undesirable, as the linter then also just accepts no 'smart' indenting/alignment at all...

**Comment by nzakas** [comment]
@oh-ren please open a separate issue for that discussion.

**Comment by oh-ren** [comment]
@nzakas Will do, was already thinking about that myself as well.

**Comment by afahim** [comment]
I've taken a stab at this, let me know if you have any comments / suggestions.
